examples/hello-world/hello-tf2/README.md

```diff
@@ -18,6 +18,23 @@ Install additional requirements:
 pip3 install tensorflow
 ```
 
+#### Notes on running with GPUs
+
+For running with GPUs, we recommend using
+[NVIDIA TensorFlow docker](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tensorflow)
+
+If you choose to run the example using GPUs, it is important to note that,
+by default, TensorFlow will attempt to allocate all available GPU memory at the start.
+In scenarios where multiple clients are involved, you have to prevent TensorFlow from allocating all GPU memory 
+by setting the following flags.
+```bash
+TF_FORCE_GPU_ALLOW_GROWTH=true TF_GPU_ALLOCATOR=cuda_malloc_async
+```
+
+If you possess more GPUs than clients, a good strategy is to run one client on each GPU.
+This can be achieved using the `-gpu` argument during simulation, e.g., `nvflare simulator -n 2 -gpu 0,1 [job]`.
+
+
```

Comment:
just a note some of the changes are already in the main branch so we will need to rebase this branch so we don't change 278 files in this PR